packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -256,13 +257,23 @@ const importPrefix = '__vite_glob_'
 
 const { basename, dirname, relative, join } = posix
 
+export interface TransformGlobImportResult {
+  s: MagicString
+  matches: ParsedImportGlob[]
+  files: Set<string>
+}
+
+/**
+ * @param optimizeExport for dynamicImportVar plugin don't need to optimize export.
+ */
 export async function transformGlobImport(
   code: string,
   id: string,
   root: string,
   resolveId: IdResolver,
-  restoreQueryExtension = false
-) {
+  restoreQueryExtension = false,
+  optimizeExport = true
```

Comment:
Why this option is needed? If no `{ import: 'foo' }` specified in the `import.meta.glob` is shouldn't affect anything.

Reply:
`import('./${a}.js?raw')` don't need to add `.then(m => m.default)`. And this options just affect this?

Reply:
Good point. Made https://github.com/vitejs/vite/pull/8071

Reply:
Do we win much by miss-aligning with the standard here? Maybe `?raw` should do this optimization and users can write `.then(m => m.default)` themselves?
Vite v3 may be a good place to introduce this as a breaking change, and then we don't need #8071. What do you think?

Reply:
I see this is a quality of life thing. When importing as `raw` or `url`, I think ppl expect it to be a string (there is nothing else in that module in the common case). I think it matches with Vite's philosophy that if 99% of users use that way we might just default to it (Plus we emit the correct type when working with `raw` and `url`). The `import: '*'` is more like an escape hatch for really advanced cases (here for example). I don't see normal users would import raw modules as modules instead of strings.